apps/web/src/providers/ConfirmProvider.tsx

```diff
@@ -1,62 +1,100 @@
-});
-
-export const useConfirm = () => useContext(ConfirmContext);
+type ConfirmState = {
+  record: Record<string, JSX.Element>;
+};
 
-export const ConfirmProvider = ({
-  children,
-}: PropsWithChildren<ConfirmContextProps>) => {
-  const [confirmRecord, setConfirmRecord] = useState<Record<string, ReactNode>>(
-    {}
-  );
-  return (
-    <ConfirmContext.Provider
-      value={{
-        confirm: ({ onCancel, onConfirm, ...props }: ConfirmProps) => {
-          return new Promise(resolve => {
-            const confirmId = String(Date.now());
-            const closeHandler = () => {
-              delete confirmRecord[confirmId];
-              setConfirmRecord({ ...confirmRecord });
-            };
-            setConfirmRecord(oldConfirmRecord => {
-              return {
-                ...oldConfirmRecord,
-                [confirmId]: (
-                  <Confirm
-                    {...props}
-                    onCancel={() => {
-                      closeHandler();
-                      onCancel?.();
-                      resolve(false);
-                    }}
-                    onConfirm={() => {
-                      closeHandler();
-                      onConfirm?.();
-                      resolve(true);
-                    }}
-                  />
-                ),
+const create = () =>
+  createStore(
+    subscribeWithSelector(
+      combine<ConfirmState, ConfirmActions>(
+        {
+          record: {},
+        },
+        (set, get) => ({
+          confirm: ({ onCancel, onConfirm, ...props }: ConfirmProps) => {
+            return new Promise(resolve => {
+              const confirmRecord = { ...get().record };
+              const confirmId = String(Date.now());
+              const closeHandler = () => {
+                delete confirmRecord[confirmId];
+                set({ record: { ...confirmRecord } });
               };
+              set(({ record }) => {
+                return {
+                  record: {
+                    ...record,
+                    [confirmId]: (
+                      <Confirm
+                        {...props}
+                        onCancel={() => {
+                          closeHandler();
+                          onCancel?.();
+                          resolve(false);
+                        }}
+                        onConfirm={() => {
+                          closeHandler();
+                          onConfirm?.();
+                          resolve(true);
+                        }}
+                      />
+                    ),
+                  },
+                };
+              });
             });
-          });
-        },
-      }}
-    >
-      {children}
-      {Object.entries(confirmRecord).map(([confirmId, confirmNode]) => {
+          },
+        })
+      )
+    )
+  );
+
+type Store = ReturnType<typeof create>;
+
+export const ConfirmContext = createContext<Store | null>(null);
+
+export const useConfirmApi = () => {
```

Comment:
What's the convention (or difference) between `useConfirmApi` and `useConfirm`? Is this a will-be-widely-used pattern?

Reply:
`useApi` retunrs 

```
{
  setState,
  getState,
  subscribe
}
```

Which is the vanilla API from javascript

https://github.com/pmndrs/zustand/blob/main/docs/guides/initialize-state-with-props.md#common-patterns